src/mscorlib/shared/System/Numerics/Vector.cs

```diff
@@ -390,7 +390,7 @@ public unsafe Vector(T[] values, int index)
             }
             if (index < 0 || (values.Length - index) < Count)
             {
-                throw new IndexOutOfRangeException();
+                throw new IndexOutOfRangeException(SR.Format(SR.Arg_InsufficientNumberOfElements, Vector<T>.Count, nameof(values)));
```

Comment:
Reintroduced this as per comment [here](https://github.com/dotnet/coreclr/commit/357e2133199f890f2a4987da85dceeb5d72ecbc8#r27916246) from @4creators in one of the previous commits...